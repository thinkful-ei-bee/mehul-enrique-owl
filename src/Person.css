.avatar {
    border-radius: 50%;
    max-height:40px;
    max-width:40px;
  }

  .location{
      margin-left:50px;
  }