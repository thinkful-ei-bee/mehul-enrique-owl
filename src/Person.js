import React from 'react';
import './Person.css';


function Person(props) {
    let onStage = props.onStage;
    let name = props.name;
    let avatar = props.avatar;
    let inSession = props.inSession;

   //inSession
   if(inSession === true && onStage === true){
    return (
        <li className="Person">
            <img className="avatar" alt="avatar-text" src={avatar}/>
            <span>{name}</span>
            <div className ="green-dot"></div>
            <p className="location">On Stage</p>
        </li>
    );
   }
   else if(inSession === true && onStage === false){
    return (
        <li className="Person">
            <img className="avatar" alt="avatar-text" src={avatar}/>
            <span>{name}</span>
            <div className ="green-dot"></div>
            <p className="location">In Session</p>
        </li>
    );
   }
   else if(inSession === false){
    return (
        <li className="Person">
            <img className="avatar" alt="avatar-text" src={avatar}/>
            <span>{name}</span>
            <div className ="grey-dot"></div>
            <p className="location">Left Session</p>
        </li>
    );
   }
  
  }

export default Person;