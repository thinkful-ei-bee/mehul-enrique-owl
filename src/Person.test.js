import React from 'react';
import ReactDOM from 'react-dom';
import Person from './Person';
import renderer from 'react-test-renderer';


it('render a onstage person without crasing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Person
    id='any'
    name='name'
    avatar='alt-text'
    inSession = {true}
    onStage = {true}
  />,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
 
});

it('snapshot test of a onStage person as expected', () => {
  const tree = renderer
    .create(<Person
      id='any'
      name='name'
      avatar='alt-text'
      inSession = {true}
      onStage = {true}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});


it('render a in session person without crasing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Person
      id='any'
      name='name'
      avatar='alt-text'
      inSession = {true}
      onStage = {false}
    />,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
   
  });
  
  it('snapshot test of a in-session person as expected', () => {
    const tree = renderer
      .create( <Person
        id='any'
        name='name'
        avatar='alt-text'
        inSession = {true}
        onStage = {false}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
  

it('render a person who left  without crasing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Person
      id='any'
      name='name'
      avatar='alt-text'
      inSession = {false}
      onStage = {false}
    />,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
   
  });
  
  it('snapshot test of a person who left as expected', () => {
    const tree = renderer
      .create(   <Person
        id='any'
        name='name'
        avatar='alt-text'
        inSession = {false}
        onStage = {false}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
  
