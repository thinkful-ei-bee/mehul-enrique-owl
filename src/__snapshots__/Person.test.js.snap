// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of a in-session person as expected 1`] = `
<li
  className="Person"
>
  <img
    alt="avatar-text"
    className="avatar"
    src="alt-text"
  />
  <span>
    name
  </span>
  <div
    className="green-dot"
  />
  <p
    className="location"
  >
    In Session
  </p>
</li>
`;

exports[`snapshot test of a onStage person as expected 1`] = `
<li
  className="Person"
>
  <img
    alt="avatar-text"
    className="avatar"
    src="alt-text"
  />
  <span>
    name
  </span>
  <div
    className="green-dot"
  />
  <p
    className="location"
  >
    On Stage
  </p>
</li>
`;

exports[`snapshot test of a person who left as expected 1`] = `
<li
  className="Person"
>
  <img
    alt="avatar-text"
    className="avatar"
    src="alt-text"
  />
  <span>
    name
  </span>
  <div
    className="grey-dot"
  />
  <p
    className="location"
  >
    Left Session
  </p>
</li>
`;
